### DISCUSSION TITLE
百度网盘视频、pdf文件无法查看

### DISCUSSION THREAD

**Discussion by x18-1** [AlistGo/alist#2181] [Q&A]
1. 通过这个软件挂在上百度网盘之后，播放百度网盘的视频会出现加载失败；
2. 打开pdf的文件会出现以下错误：
```
PDF.js v3.0.44 (build: 6852001b0)
Message: Failed to fetch
```

**当我使用阿里云盘的时候没有出现这些问题**

**Comment by DEKVIW** [comment]
同样，百度网盘播放视频提示重连,重连五次以后失败；测试了下，视频不超过50兆是可以播放的，超过50兆就会重连五次然后无法播放。而且百度网盘好多格式的都打不开，阿里还好常用的都好用

**Comment by 8uzzlightyear** [comment]
我也是这种情况，百度网盘视频预览不了。我用alist就是受不了百度网盘自带的播放器看视频ahh

**Comment by xixia123** [comment]
我也是没有办法看百度网盘里面的视频，好难受！！！

**Comment by changshanzhao** [comment]
同，在线求解

**Comment by earor-R** [comment]
我出了上述各位的问题以外，还不能下载，可是我昨天刚部署完还好好的

**Reply by earor-R** [reply]
我错了，并不是昨天好好的，而是有的能打开且能下载，有的不能打开且不能下载。昨天我恰好打开了能打开的。好像是自己上传的视频几乎都打不开，但是分享的电影之类的传播比较广的都能打开...

**Comment by leekule** [comment]
我百度网盘有的文件夹里的文件还不能修改文件名。看大家百度网盘挂载这么多问题，以后会考虑放弃充百度网盘的超级vip会员了。

**Comment by choinage** [comment]
使用本地代理（如果只在本地使用的话）即可
![image](https://user-images.githubusercontent.com/88093924/212677515-83ac3d68-1b6f-4e5f-b3db-356582d5540c.png)

**Comment by astoaq** [comment]
使用本地代理也不行

**Comment by hiwens** [comment]
在alist主页上打开百度网盘word、excel、ppt、PDF等文件基本上都是无法查看，打开PDF就是出现
```
PDF.js v3.0.44 (build: 6852001b0)
信息：network error
```
打开word、excel就是出现
```
An error occurred
We're sorry, but for some reason we can't open this for you.
```
但是视频可以正常观看。
raidrive挂载后文件管理打开，一直在加载直到未响应。
如何解决？

**Reply by heli9913** [reply]
同问

**Reply by Mizar29** [reply]
同问

**Reply by wowo14735** [reply]
我用的谷歌云盘也是这样。

**Reply by ET-FJ** [reply]
同问

**Comment by hioTEC** [comment]
应该是作者部署的pdf.js挂了，将Alist管理-设置-预览-iframe预览中的“pdf”改为google的预览器即可，将原来的pdf.js那行换成这个：
"Google":"https://docs.google.com/gview?url=$e_url&embedded=true"

**Reply by ghost** [reply]
还是不可以

**Reply by hioTEC** [reply]
这就不清楚了，可能有别的问题……我改完就ok了，word预览我也把微软那个删了只留的Google这个

**Reply by ricolxwz** [reply]
我用这个方法改完之后就可以用了, 应该是pdf.js自身的问题.

**Reply by lizhiliao** [reply]
![屏幕截图 2024-09-23 032157](https://github.com/user-attachments/assets/1a6c3a6b-1617-4875-b436-004c961c12c5)

换了也不行

**Reply by lizhiliao** [reply]
> ![屏幕截图 2024-09-23 032157](https://private-user-images.githubusercontent.com/63957470/369712288-1a6c3a6b-1617-4875-b436-004c961c12c5.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwMzMyNjEsIm5iZiI6MTcyNzAzMjk2MSwicGF0aCI6Ii82Mzk1NzQ3MC8zNjk3MTIyODgtMWE2YzNhNmItMTYxNy00ODc1LWI0MzYtMDA0Yzk2MWMxMmM1LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIyVDE5MjI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzMDE4ODIyZWM2YzZlZjE0MDJhOTg5Y2Y1OTU4YzkyNjg0ZTljZDgyNDI5OTQ0OTc2MTc4NDgwNDMzYTYxMjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.di8Ie76QcNKz_30oY3iPcfHwVWeaSYPk7t2Ub9rUJ4c)
> 
> 换了也不行

链接里的$e_url换成$durl就能正常预览了，不过需要使用翻墙工具，要是有国内网络能用的pdf预览器就好了

**Reply by ricolxwz** [reply]
不是pdf.js的问题，是bucket没有开CORS。

**Comment by Mizar29** [comment]
我也遇到这个问题了，请问解决了吗

**Comment by heli9913** [comment]
查看应该没有问题，如果文件太大了可能不行，百度云盘还是有问题的不推荐

--------------------------------
CAA-heli；QQ:2630334956；TE：057188297134


----- 原始邮件 -----
发件人：Alcor ***@***.***>
收件人：alist-org/alist ***@***.***>
抄送人：heli9913 ***@***.***>,  Comment ***@***.***>
主题：Re:_[alist-org/alist]_百度网盘视频、pdf文件无法查看_(Discussion_#2181)
日期：2023年03月23日 10点36分


我也遇到这个问题了，请问解决了吗

**Comment by heli9913** [comment]
百度盘还是有问题的，建议用阿里
----- 原始邮件 -----
发件人：Alcor ***@***.***>
收件人：alist-org/alist ***@***.***>
抄送人：heli9913 ***@***.***>,  Comment ***@***.***>
主题：Re:_[alist-org/alist]_百度网盘视频、pdf文件无法查看_(Discussion_#2181)
日期：2023年03月23日 10点34分


同问

&mdash;

**Comment by AkarinLiu** [comment]
我也是这样！
```
{"error_code":31326,"error_msg":"user is not authorized, hitcode:119","request_id":443353246343788909}
```

**Comment by LONGZR007** [comment]
我的pdf不能看，从http改https好了，不过ppt还是不能看

**Comment by chinazzb** [comment]
有大佬解决一下吗？ 错误信息如下图视频正常观看，用阿里云盘的获取到的资源是https，百度网盘是http，我看网上说是chrome做了限制，但设置了浏览器的“不安全内容”为允许
![1708160586538](https://github.com/alist-org/alist/assets/28751302/b46f5c4d-ff9d-4ce6-9f43-5f4c86484186)
![image](https://github.com/alist-org/alist/assets/28751302/29c8f1cb-cd87-4b53-987c-2fbb6a570933)

大佬可以参考一下内容
https://www.cnblogs.com/goloving/p/14780408.html

**Comment by AruoTan** [comment]
PDF 无法查看的，有可能是 IDM 捕获了 PDF，导致 PDF.js 无法获得资源。可以尝试在 IDM 中的自动捕获类型中去掉 PDF 类型以解决问题。

**Reply by F96371** [reply]
不知道如何解决：
请问我下载pdf后打开显示:
![image](https://github.com/user-attachments/assets/86618456-6503-4fc8-894f-49302d172867)
下载pptx打开后显示：
![image](https://github.com/user-attachments/assets/49a00638-b21d-4205-9c76-092b7050f5b7)
![image](https://github.com/user-attachments/assets/8dd70fd7-7371-44e7-9da5-98a76dd90018)

**Reply by F96371** [reply]
解决办法，下载路径切换成官方。
